### PR TITLE
PISTON-507: allow specifying a body for sendevent NOTIFY

### PR DESCRIPTION
**Pull request by danielfinke**

### CODE REVIEW

File: applications/ecallmgr/src/freeswitch.erl [hunk 1]
```diff
@@ -28,6 +28,7 @@
         ]).
 -export([nixevent/2]).
 -export([sendevent/3
+        ,sendevent/5
```
Comment (lazedo) [inline applications/ecallmgr/src/freeswitch.erl]: no need for this, filter the Headers you want to send

File: applications/ecallmgr/src/freeswitch.erl [hunk 2]
```diff
@@ -109,7 +110,19 @@ event(Node, Events, Timeout) -> ?FS_MODULE:event(Node, Events, Timeout).
...
-sendevent(Node, EventName, Headers) -> ?FS_MODULE:sendevent(Node, EventName, Headers).
+-spec sendevent(atom(), atom(), list(), api_binary(), api_binary()) -> 'ok'.
+sendevent(Node, EventName, Headers) ->
+    sendevent(Node, EventName, Headers, 'undefined', 'undefined').
+
+sendevent(Node, EventName, Headers, 'undefined', _) ->
+    ?FS_MODULE:sendevent(Node, EventName, Headers);
+sendevent(Node, EventName, Headers, Body, ContentType) ->
+    Headers1 = [{"body", Body}
+               ,{"content-length", kz_term:to_list(byte_size(Body))}
```
Comment (lazedo) [inline applications/ecallmgr/src/freeswitch.erl]: no need for `content-length` as freeswitch will calculate it

File: applications/ecallmgr/src/ecallmgr_fs_notify.erl [hunk 3]
```diff
@@ -124,23 +129,27 @@ notify(Username, Realm, Event) ->
...
     Headers = [{"profile", ?DEFAULT_FS_PROFILE}
               ,{"contact-uri", Contact}
               ,{"extra-headers", SIPHeaders}
               ,{"to-uri", To}
               ,{"from-uri", From}
               ,{"event-string", Event}
               ],
-    Resp = freeswitch:sendevent(Node, 'NOTIFY', Headers),
-    lager:info("send NOTIFY with Event '~s' to '~s@~s' via ~s: ~p", [Event, Username, Realm, Node, Resp]).
+    Body = kz_json:get_binary_value(<<"Body">>, JObj),
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: `get_ne_binary_value` to make sure is not empty

File: applications/ecallmgr/src/ecallmgr_fs_notify.erl [hunk 4]
```diff
@@ -124,23 +129,27 @@ notify(Username, Realm, Event) ->
...
+    ContentType = kz_json:get_binary_value(<<"Content-Type">>, JObj),
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: same here for get_ne_binary_value

File: applications/ecallmgr/src/ecallmgr_fs_notify.erl [hunk 5]
```diff
@@ -124,23 +129,27 @@ notify(Username, Realm, Event) ->
...
+    Resp = freeswitch:sendevent(Node, 'NOTIFY', Headers, Body, ContentType),
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: as mentioned below, there is no nered to add `freeswitch:sendevent/5`, filter the headers and send

**Comment by danielfinke** [comment]
@lazedo changes made

**Comment by jamesaimonetti** [comment]
@danielfinke thanks, merged. Trying to get movement on the other PRs too; today or tomorrow certainly.

**Comment by jamesaimonetti** [comment]
Dangit, ref'd the wrong PR; its commit 87dcc8bc61ccbf652feaf5a4a4dd6ecf0c239a20 in master

**Pull request closed without merging**